scalableminds/webknossos
# Disable editing of super voxel skeletons in skeleton mode

- This PR introduces a tree type property to be able to distinguish between agglomerate skeletons and user-created trees. This property is then used to:
  - prevent (node/edge-related) modifications of agglomerate skeletons in general
  - only allow to add or delete edges of agglomerate skeletons if the proofreading tool is active
  - inform users to use the proofreading tool if they are trying to modify agglomerate skeletons while the proofreading tool is not active
  
### TODO:
- [x] Backend ToDos, see https://github.com/scalableminds/webknossos/pull/7086#issuecomment-1570367457
- [x] Include tree types in NML
- [x] Should agglomerate skeletons be visually distinguished from user-created skeletons somehow?
- ~~[ ] [Optional] Allow to convert agglomerate skeletons to freely modifiable trees~~

### URL of deployed dev instance (used for testing):
- https://proofreadingskeletonmodification.webknossos.xyz

### Steps to test:
- Create an annotation on a dataset with an agglomerate view (e.g. test-agglomerate-file)
- Create a skeleton by clicking some nodes
- Load an agglomerate skeleton
  - Try to delete a node or add new nodes -> Nothing should happen and an error toast should be shown
  - Try to delete or add an edge -> Nothing should happen and an error toast hinting at the proofreading tool should be shown
- Switch to the proofreading tool
  - Try to delete a node or add new nodes -> Nothing should happen and an error toast should be shown
  - Try to delete or add an edge -> Should work as before
- Reload the annotation. The tree types should be restored.
- Export and import an NML using the backend and frontend functionality. Non-default tree types should, and default tree types can, be serialized to NML.

### Issues:
- fixes #6935 

------
(Please delete unneeded items, merge only when none are left open)
